lib/engines/content_object_store/test/integration/content_block_editions_test.rb

```diff
@@ -25,26 +26,36 @@ class ContentBlockEditionsTest < ActionDispatch::IntegrationTest
     document_title = "Some Title"
     block_type = "block_type"
     details = {
-      foo: "Foo text",
-      bar: "Bar text",
+      "foo" => "Foo text",
+      "bar" => "Bar text",
     }
 
-    ContentObjectStore::ContentBlockSchema.expects(:find_by_block_type).with(block_type).returns(schema)
-    ContentObjectStore::ContentBlockEdition.expects(:create!).with do |args|
-      args.to_h == {
-        document_title:,
-        block_type:,
-        details:,
-      }.with_indifferent_access
+    # This UUID is created by the database so instead of loading the record
+    # we stub the initial creation so we know what UUID to check for.
+    ContentObjectStore::ContentBlockEdition.any_instance.stubs(:create_random_id)
+      .returns(@content_id)
+
+    ContentObjectStore::ContentBlockSchema.stubs(:find_by_block_type).with(block_type).returns(schema)
+    ContentObjectStore::ContentBlockSchema.stub_const(:VALID_SCHEMAS, [block_type]) do
+      assert_changes -> { ContentObjectStore::ContentBlockDocument.count }, from: 0, to: 1 do
+        assert_changes -> { ContentObjectStore::ContentBlockEdition.count }, from: 0, to: 1 do
```

Comment:
Oooh, I didn't realise you could nest these expectations - nice!

Reply:
My brain is still switching over from RSpec. I'm curious to know if there's a better way? Because i'm also making an assumption that we start on 0 which isn't perfect because there might be some seeding that adds some in one day.. I paused from going this far though:

```ruby
original_count = ContentObjectStore::ContentBlockDocument.count
assert_changes -> { ContentObjectStore::ContentBlockDocument.count }, from: original_count, to: original_count + 1 do
```